funcx_endpoint/funcx_endpoint/endpoint/endpoint.py

```diff
@@ -559,3 +563,49 @@ def print_endpoint_table(conf_dir: pathlib.Path, ofile=None):
         table._width[idx_name] = max(10, table._width[idx_name])
 
         print(table.draw(), file=ofile)
+
+    @staticmethod
+    def get_metadata(config: Config) -> dict:
+        metadata: dict = {}
+
+        metadata["endpoint_version"] = __version__
+        metadata["hostname"] = socket.getfqdn()
+
+        # should be more accurate than `getpass.getuser()` in non-login situations
+        metadata["local_user"] = pwd.getpwuid(os.getuid()).pw_name
+
+        # the following are read from the HTTP request by the web service, but can be
+        # overridden here if desired:
+        metadata["ip_address"] = None
+        metadata["sdk_version"] = None
+
+        expand_list = ["strategy", "provider", "launcher"]
+
+        def to_dict(o):
+            mems = {"_type": type(o).__name__}
+
+            for k, v in o.__dict__.items():
+                if k.startswith("_"):
+                    continue
+
+                if k in expand_list:
+                    mems[k] = to_dict(v)
+                elif not isinstance(v, str):
+                    mems[k] = repr(v)
+                else:
+                    mems[k] = v
+
+            return mems
+
+        try:
+            metadata["config"] = to_dict(config)
+            metadata["config"]["executors"] = [
```

Comment:
Let's add a comment here that we don't need to support multiple executors long-term. The plan is to have one EP be just one executor.